compiler/rustc_middle/src/mir/interpret/error.rs

```diff
@@ -502,4 +503,19 @@ impl InterpError<'_> {
             _ => false,
         }
     }
+
+    /// Should this error be reported as a hard error, preventing compilation, or a soft error,
+    /// causing a deny-by-default lint?
+    pub fn is_hard_err<'tcx>(&self, tcx: TyCtxt<'tcx>, def: DefId) -> bool {
+        use InterpError::*;
+        if let Some(def) = def.as_local() {
+            // (Associated) consts only emit a lint, since they might be unused.
+            !matches!(tcx.def_kind(def), DefKind::Const | DefKind::AssocConst)
```

Comment:
Sorry, this is not what I meant. That weird assoc const logic, and the cross-crate logic, should *not* be here in this file.

I literally meant exactly what I did in https://github.com/rust-lang/rust/pull/86194: just a simple match on the `InterpError`.